bigtable-hbase-replication/src/main/java/com/google/cloud/bigtable/hbase/replication/adapters/BigtableWALEntry.java

```diff
@@ -0,0 +1,30 @@
+package com.google.cloud.bigtable.hbase.replication.adapters;
+
+import org.apache.hadoop.hbase.Cell;
+import org.apache.hadoop.hbase.wal.WAL;
+
+import java.util.ArrayList;
+/**
+ * BigtableWALEntry abstracts minimal functionality from WAL.Entry required for this replication library.
+ */
+public class BigtableWALEntry {
+    private long timeStamp;
+    private ArrayList<Cell> cells;
+    private String tableName;
+
+    public BigtableWALEntry(long timeStamp, ArrayList<Cell> cells, String tableName) {
```

Comment:
Exactly ! call this constructor from the 1.x and 2.x endpoints and you dont' need the IMPL files.